lib/fog/compute/kubevirt.rb

```diff
@@ -84,28 +117,123 @@ def initialize(options={})
           # Kubeclient needs different client objects for different API groups. We will keep in this hash the
           # client objects, indexed by API path/version.
           @clients = {}
+        end
 
-          @client = kubevirt_client()
+        def virt_supported?
+          virt_enabled = false
 
-          # TODO expect a specific token
-          @oc_client = openshift_client()
-          @kube_client = kube_client()
+          begin
+            api_versions = kubevirt_client.api["versions"]
+            api_versions.each do |ver|
+              if ver["groupVersion"]&.start_with?(KUBEVIRT_GROUP)
+                virt_enabled = true
+              end
+            end
+          rescue => err
+            # we failed to communicate or to evaluate the version format
+            $log.warn("Failed to detect kubevirt on provider with error: #{err.message}")
+          end
+
+          virt_enabled
         end
 
-        private
+        def valid?
+          kube_client.api_valid?
+        end
 
-        def client
-          @client
+        class WatchWrapper
+
+          def initialize(watch, mapper)
+            @watch = watch
+            @mapper = mapper
+          end
+
+          def each
+            @watch.each do |notice|
```

Comment:
we'd might want to filter out notices before using them according to: 

https://github.com/ManageIQ/manageiq-providers-kubevirt/blob/master/app/models/manageiq/providers/kubevirt/infra_manager/refresh_worker/runner.rb#L136

Also we should parse only the latest one, the previous once are considered as obsolete from the provider's pov.

Reply:
We filter them in the provider. I would keep this code as simple as possible.